pkg/scheduler/algorithm/predicates/metadata.go

```diff
@@ -371,45 +370,46 @@ func getTPMapMatchingExistingAntiAffinity(pod *v1.Pod, nodeInfoMap map[string]*s
 		allNodeNames = append(allNodeNames, name)
 	}
 
-	var lock sync.Mutex
-	var firstError error
-
-	topologyMaps := newTopologyPairsMaps()
-
-	appendTopologyPairsMaps := func(toAppend *topologyPairsMaps) {
-		lock.Lock()
-		defer lock.Unlock()
-		topologyMaps.appendMaps(toAppend)
-	}
-	catchError := func(err error) {
```

Comment:
if we set the buffer of this channel to zero, then we can make this non-blocking as follows:

```
catchError := func(err error) {
 select {
    case errCh <- err:
    default:
    }
}
```

Reply:
Sounds good.